gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/FsBenchmark.java

```diff
@@ -0,0 +1,468 @@
+
+                byte[] readBuffer = new byte[readSize];
+
+                initLatch.countDown();
+                awaitUninterruptibly(startLatch);
+                try {
+                  for (int j = 0; j < numReads; j++) {
+                    try (FSDataInputStream input = fs.open(testFile)) {
+                      long readStart = System.nanoTime();
+                      long fileBytesRead = 0;
+                      int bytesRead;
+                      do {
+                        long readCallStart = System.nanoTime();
+                        bytesRead = input.read(readBuffer);
+                        if (bytesRead > 0) {
+                          fileBytesRead += bytesRead;
+                          readCallBytes.accept(bytesRead);
+                        }
+                        readCallTimeNs.accept(System.nanoTime() - readCallStart);
+                      } while (bytesRead >= 0);
+
+                      readFileBytes.accept(fileBytesRead);
+                      readFileTimeNs.accept(System.nanoTime() - readStart);
+                    }
+                  }
+                } finally {
+                  stopLatch.countDown();
+                }
+                return null;
+              }));
+    }
+    executor.shutdown();
+
+    awaitUninterruptibly(initLatch);
+    long startTimeNs = System.nanoTime();
+    startLatch.countDown();
+    awaitUninterruptibly(stopLatch);
+    long runtimeNs = System.nanoTime() - startTimeNs;
+
+    // Verify that all threads completed without errors
+    futures.forEach(Futures::getUnchecked);
+
+    printTimeStats("Read call time", readCallTimeNsList);
+    printSizeStats("Read call data", readCallBytesList);
+    printThroughputStats("Read call throughput", readCallTimeNsList, readCallBytesList);
+
+    printTimeStats("Read file time", readFileTimeNsList);
+    printSizeStats("Read file data", readFileBytesList);
+    printThroughputStats("Read file throughput", readFileTimeNsList, readFileBytesList);
+
+    System.out.printf(
+        "Read average throughput (MiB/s): %.3f%n",
+        bytesToMebibytes(combineStats(readFileBytesList).getSum()) / nanosToSeconds(runtimeNs));
+  }
+
+  private int benchmarkRandomRead(FileSystem fs, Map<String, String> args) {
+    if (args.size() < 1) {
+      System.err.println(
+          "Usage: random-read"
+              + " --file=gs://${BUCKET}/path/to/test/object"
+              + " [--num-open=<number of file open>]"
+              + " [--read-size=<read buffer size in bytes>]"
+              + " [--num-reads=<number of random reads per file open>]"
+              + " [--num-threads=<number of threads to run test>]");
+      return 1;
+    }
+
+    Path testFile = new Path(args.get("--file"));
+
+    warmup(
+        args,
+        () ->
+            benchmarkRandomRead(
+                fs,
+                testFile,
+                /* numOpen= */ 5,
```

Comment:
How are these number chosen? (Asking bcuz it looks very differently than `read`)

Reply:
Just something that does not run for too long to not make benchmarks too long. Read test always reads whole file, so there less room to reduce warmup phase time.